### PR TITLE
✨  [minor] Add deployment issues into Queue crd

### PR DESCRIPTION
**Pull request by Pohfy123**
**What's the change or impact of this PR?**:
  - Categorize deployment issues and attach into Queue crd
  - Send report with deployment issues when staging/active promotion failed

**Related issue link**:

### CODE REVIEW

**Review by sunny299 — APPROVED** [review]

File: internal/staging/collect_result.go [hunk 1]
```diff
@@ -453,3 +481,200 @@ func (c *controller) getReverificationStatus(queue *s2hv1beta1.Queue) rpc.Compon
...
+					}
+
+					if found {
+						break
+					}
+
+					runningState := containerStatus.State.Running
+					if runningState != nil {
+						// if running 0/1, count as CrashLoopBackOff
+						if containerStatus.RestartCount > 0 {
```
Comment (quraynai) [inline internal/staging/collect_result.go:590]: If component is in running 0/1. and restartcount == 0. should we handle it?

Comment (Pohfy123) [reply internal/staging/collect_result.go:590]: it will be treated as undefined type, since means that readiness failed but liveness is good

Comment (Pohfy123) [reply internal/staging/collect_result.go:590]: and it might mean that it cannot be started in time without any issues

Comment (Pohfy123) [reply internal/staging/collect_result.go:592]: @quraynai oh, I haven't handled it yet, will update

**Review by sunny299 — APPROVED** [review]

**Pull request merged**